test/one/profiler/test/TestProcess.java

```diff
@@ -0,0 +1,302 @@
+        }
+    }
+
+    @Override
+    public void close() {
+        p.destroy();
+
+        try {
+            waitForExit(p, 20);
+        } catch (TimeoutException e) {
+            log.log(Level.WARNING, "Failed to terminate child process", e);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        } finally {
+            if (testName != null) {
+                moveLogs();
+            }
+            clearTempFiles();
+        }
+    }
+
+    public long pid() {
+        try {
+            return (long) pid.invokeExact(p);
+        } catch (Throwable e) {
+            throw new IllegalStateException("Unsupported API", e);
+        }
+    }
+
+    public Output waitForExit(String fileId) throws TimeoutException, InterruptedException, IOException {
+        waitForExit(p, timeout);
+        return readFile(fileId);
+    }
+
+    private void waitForExit(Process p, int seconds) throws TimeoutException, InterruptedException {
+        if (!p.waitFor(seconds, TimeUnit.SECONDS)) {
+            p.destroyForcibly();
+            throw new TimeoutException("Child process has not exited");
+        }
+    }
+
+    public Output profile(String args) throws IOException, TimeoutException, InterruptedException {
+        return profile(args, false);
+    }
+
+    public Output profile(String args, boolean sudo) throws IOException, TimeoutException, InterruptedException {
+        List<String> cmd = new ArrayList<>();
+        if (sudo) {
+            cmd.add("/usr/bin/sudo");
+        }
+        cmd.add("build/bin/asprof");
+        addArgs(cmd, args);
+        cmd.add(Long.toString(pid()));
+        log.log(Level.FINE, "Profiling " + cmd);
+
+        Process p = new ProcessBuilder(cmd)
+                .redirectOutput(createTempFile("%pout"))
+                .redirectError(createTempFile("%perr"))
+                .start();
+
+        waitForExit(p,10);
+        int exitCode = p.waitFor();
+        if (exitCode != 0) {
+            throw new IOException("Profiling call failed " + readPOut().toString() + readPErr().toString());
+        }
+
+        return readPOut();
+    }
+
+    public Output readFile(String fileId) {
+        File f = tmpFiles.get(fileId);
+        try (Stream<String> stream = Files.lines(f.toPath())) {
+            return new Output(stream.toArray(String[]::new));
+        } catch (IOException e) {
+            return new Output(new String[0]);
+        }
+    }
+
+    public Output readPOut() throws IOException {
+        return readFile("%pout");
+    }
+
+    public Output readPErr() throws IOException {
+        return readFile("%perr");
+    }
+
+    public Output readOut() throws IOException {
+        return readFile("%out");
+    }
+
+    public Output readErr() throws IOException {
+        return readFile("%err");
+    }
+
+    public Output readF() throws IOException {
+        return readFile("%f");
+    }
```

Comment:
`readXX` methods have unclear names and they just do nothing useful.
I'd leave only one `readFile` method and probably extract `%pout` and other similar strings to constants.

Reply:
Addressed. Will reflect in next commit